server/src/test/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerTests.java

```diff
@@ -68,14 +68,16 @@ public void testThatInstancesAreCachedAndReused() {
         assertSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.CURRENT),
                 PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.CURRENT));
         // same es version should be cached
-        assertSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.V_6_2_1),
-                PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.V_6_2_1));
-        assertNotSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.V_6_0_0),
-                PreBuiltAnalyzers.STANDARD.getAnalyzer(Version.V_6_0_1));
+        Version versionA = randomVersion(random());
+        Version versionB = randomValueOtherThan(versionA, () -> randomVersion(random()));
+        assertSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA),
+                PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA));
+        assertNotSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA),
+                PreBuiltAnalyzers.STANDARD.getAnalyzer(versionB));
 
         // Same Lucene version should be cached:
-        assertSame(PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_6_2_1),
-            PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_6_2_2));
+        assertSame(PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_7_2_0),
```

Comment:
This is going to eventually break when 8.0 has lucene upgraded. I think you could do this instead with 2 hardcoded different versions that are old, eg `Version.fromString("5.0.0")` and `Version.fromString("5.0.1")`, which will use the oldest version of lucene we use in Version.java.

Reply:
I see. good catch!